tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -633,16 +636,17 @@ private void OnRoutingEndpointMatched(object arg)
                     return;
                 }
 
-                RouteEndpoint? endpoint = null;
+                Endpoint? endpoint = null;
+                RoutePattern? routePattern = null;
 
                 if (rawEndpointFeature.TryDuckCast<EndpointFeatureProxy>(out var endpointFeatureInterface))
                 {
```

Comment:
I think we can simplify the method from this point. `Endpoint` isn't very useful to us in practical terms, so I think we only need to keep the `if (endpointObj.TryDuckCast<RouteEndpoint>(out var routeEndpoint))` from the `ExtractEndpoint` method.

If it _isn't_ a `RouteEndpoint`, then there won't be a route to extract, or tags to set, so we can bypass all the extra work in this method. I could be missing something that needs setting, so worth double checking, but I think we can do a quick return, e.g. something like.

```csharp
object endpointObject = null;

if (rawEndpointFeature.TryDuckCast<EndpointFeatureProxy>(out var endpointFeatureInterface)
{
    endpointObject = endpointFeatureInterface.GetEndpoint();
}

if (endpoint is null && rawEndpointFeature.TryDuckCast<EndpointFeatureStruct>(out var endpointFeatureStruct))
{
    endpoint = endpointFeatureStruct.Endpoint;
}

if (!endpointObject.TryDuckCast<RouteEndpoint>(out var endpoint))
{
    // Not a RouteEndpoint, could be an Endpoint from 405
    return;
}
```

Reply:
Very true thanks, all simplified in 9e02c49, basically almost like before but still returning object instead of RouteEndpoint in `EndpointFeatureStruct ` and `EndpointFeatureProxy ` and later trying duck casting to RouteEndpoint and returning if null like suggested.